### PR TITLE
#484. Lock saving enriched judgment

### PR DESCRIPTION
**Pull request by madryk**

### CODE REVIEW

File: saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/apply/JudgmentEnrichmentServiceIntTest.java [hunk 1]
```diff
@@ -40,9 +45,15 @@
...
     
+    @Autowired
+    private JudgmentRepository judgmentRepository;
+    
     
     private Judgment judgment;
     
+    @Autowired
+    private EntityManager entityManager;
+
```
Comment (lukdumi) [inline saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/apply/JudgmentEnrichmentServiceIntTest.java]: Would better if it was above judgment (@Autowired services grouped together)

File: saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/apply/JudgmentEnrichmentService.java [hunk 2]
```diff
@@ -48,11 +51,26 @@
             return null;
         }
         
+        entityManager.detach(judgment);
```
Comment (lukdumi) [inline saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/apply/JudgmentEnrichmentService.java]: Please, add a comment above the method that the returned judgment will be in a detached state and explain why it is necessary?

File: saos-persistence/src/main/java/pl/edu/icm/saos/persistence/common/GeneratedEntityMergeException.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package pl.edu.icm.saos.persistence.common;
+
+/**
+ * Exception thrown when attempting to merge generated intity into hibernate session
```
Comment (lukdumi) [inline saos-persistence/src/main/java/pl/edu/icm/saos/persistence/common/GeneratedEntityMergeException.java]: entity

File: saos-persistence/src/main/java/pl/edu/icm/saos/persistence/common/GeneratedEntityMergeException.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ */
+public class GeneratedEntityMergeException extends RuntimeException {
+
+    private static final long serialVersionUID = 7156528396044589039L;
+
+    
+    //------------------------ CONSTRUCTORS --------------------------
+    
+    public GeneratedEntityMergeException(Object entity) {
+        super("Detected attempt to merge generated entity " + entity + ". Merging generated entities is not allowed");
```
Comment (lukdumi) [inline saos-persistence/src/main/java/pl/edu/icm/saos/persistence/common/GeneratedEntityMergeException.java]: ... merge <b>a</b> generated.... Merging <b>of</b>...

File: saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/apply/JudgmentEnrichmentServiceIntTest.java [hunk 5]
```diff
@@ -72,6 +83,40 @@ public void findOneAndEnrich() {
...
+    
+    
+    @Test(expected = GeneratedEntityMergeException.class)
+    public void findOneAndEnrich_GENERATED_SAVE_EXCEPTION() {
+        
+        // given
+        Judgment enrichedJudgment = judgmentEnrichmentService.findOneAndEnrich(judgment.getId());
+        
+        // execute
+        judgmentRepository.save(enrichedJudgment);
```
Comment (lukdumi) [inline saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/apply/JudgmentEnrichmentServiceIntTest.java]: I wonder what can be the result of saving not judgment but one of its generated referenced regulations.

File: saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/apply/JudgmentEnrichmentServiceIntTest.java (hunk 1 above)
Comment (madryk) [inline saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/apply/JudgmentEnrichmentServiceIntTest.java]: :+1:

File: saos-persistence/src/main/java/pl/edu/icm/saos/persistence/common/GeneratedEntityMergeException.java (hunk 3 above)
Comment (madryk) [inline saos-persistence/src/main/java/pl/edu/icm/saos/persistence/common/GeneratedEntityMergeException.java]: :+1:

File: saos-persistence/src/main/java/pl/edu/icm/saos/persistence/common/GeneratedEntityMergeException.java (hunk 4 above)
Comment (madryk) [inline saos-persistence/src/main/java/pl/edu/icm/saos/persistence/common/GeneratedEntityMergeException.java]: ok

File: saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/apply/JudgmentEnrichmentService.java (hunk 2 above)
Comment (madryk) [inline saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/apply/JudgmentEnrichmentService.java]: ok

File: saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/apply/JudgmentEnrichmentServiceIntTest.java (hunk 5 above)
Comment (madryk) [inline saos-enrichment/src/test/java/pl/edu/icm/saos/enrichment/apply/JudgmentEnrichmentServiceIntTest.java]: ok, i will add it.

**Pull request closed without merging**